src/Stratis.Bitcoin.Features.SmartContracts/Wallet/SmartContractTransactionService.cs

```diff
@@ -216,15 +212,9 @@ private string[] ReplaceSignatures(string[] parameters, string[] signatures)
                 {
                     if (encodedSigs == null)
                     {
-                        var sigs = (signatures ?? new string[0]).Select(s => Convert.FromBase64String(s)).ToArray();
-                        if (sigs.Any(s => s.Length != signatureLength || s[0] < minHeaderByte || s[0] > maxHeaderByte))
-                            throw new Exception("Invalid signature(s).");
-
-                        var sigbuf = new byte[sigs.Length * signatureLength];
-                        for (int j = 0; j < sigs.Length; j++)
-                            Array.Copy(sigs[j], 0, sigbuf, j * signatureLength, signatureLength);
-
-                        encodedSigs = $"{(int)MethodParameterDataType.ByteArray}#{Encoders.Hex.EncodeData(sigbuf)}";
+                        var serializer = new ContractPrimitiveSerializer(this.network);
+                        var sigbuf = serializer.Serialize(signatures ?? new string[0]);
+                        encodedSigs = $"{(int)MethodParameterDataType.ByteArray}#{BitConverter.ToString(sigbuf).Replace("-","")}";
```

Comment:
Can you assign the result to parameters[i] and put break after it ? This will simplify code bit

Reply:
@YakupIpek , i've decided to refactor this to make arrays more generally usable outside of just the signature passing use-case.